File: c/euromilhoes.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <stdbool.h>
#include <time.h>

int compare_values(int *values, int *generated_values, int size){
    int cnt = 0;
    for(int i=0; i<size; i++){
        for(int a=0; a<size; a++){
            if(values[i]==generated_values[a]){
                cnt++;
            }
        }
    }
    return cnt;
}

bool check_genarated_values(int *values, char size, int max){
    srand(time(NULL));
    for(int i=0; i<size; i++){
        for(int a=0; a<size; a++){
            if(values[i]==values[a] && i!=a){
                values[i] = rand() % max + 1;
                return false;
            }
        }
    }
    return true;
}


void generate_values(int* values, char size, int max){
    bool x = false;
    srand(time(NULL));
    for(int i=0; i<size; i++)
        values[i] = rand() % max + 1; 
        
    while(x==false)
        x=check_genarated_values(values, size, max);
        
    for(int i=0; i<size; i++){
        if(size==5 && i==0)
            printf("\ngenerated nums: %d, ", values[i]);
        else if(size==2 && i==0)
            printf("\ngenerated stars: %d, ", values[i]);
        else
            printf("%d, ", values[i]);
    }
}

bool check_values(int flag,int size, int *values){
    for(int i=0; i<size; i++){
        for(int a=0; a<size; a++){
            if(values[i]==values[a] && i!=a){
                //printf("same number");
                printf("equal value, change the value %d: ", values[i]);
                scanf("%d", &values[i]);
                return false;
            }
            
            if((values[i]>51 ||values[i]<1) && flag==1 || (values[i]>11 ||values[i]<1) && flag==0){
                printf("value %d is invalid, change: ", values[i]);
                scanf("%d", &values[i]);
                return false;
            }
        }
    }
    return true;
}

void read_values(int *nums_stars,char flag){
    char values[20], c=1;
    
    if(flag==1)
        printf( "Enter the numbers :");
    else if(flag==0)
        printf("Enter the stars: ");
    scanf("%s", values);
    
    char * token = strtok(values, ",");
    nums_stars[0]=atoi(token);
    
    while( token != NULL ) {
      //printf( " %s\n", token ); //printing each token
      token = strtok(NULL, ",");
      
      if(token != NULL){    
        nums_stars[c]=atoi(token);
        c++;}
    }
}

int main()
{
   int nums[5], stars[2], generated_nums[5], generated_stars[2];
   bool check=false;
   read_values(nums, 1); 
   while(check==false)
        check=check_values(1, 5, nums);
        
   read_values(stars, 0);
   while(check==false)
        check=check_values(0, 2, stars);
        
   generate_values(generated_nums, 5, 50);
   generate_values(generated_stars, 2, 10);
   int equal_nums = compare_values(nums, generated_nums, 5);
   int equal_stars = compare_values(stars, generated_stars, 2);
   printf("\nnumbers equals: %d, stars equals: %d", equal_nums, equal_stars);
   return 0;
}

File: python/euromilhoes.py
import random

class euro:
    def __init__(self):
        self.nums = [0,0,0,0,0]
        self.stars = [0,0]
        self.generated_nums = [0,0,0,0,0]
        self.generated_stars = [0,0]
    
    def check_values(self, max, values):
        for index, x in enumerate(values):
            if x > str(max) or x < "1":
                if max==50:
                    self.nums[index] = input(f"invalid num {self.nums[index]}, change: ")
                elif max==10:
                    self.stars[index] = input(f"invalid star {self.stars[index]}, change: ")
                return False
        return True

    def check_repeated_values(self, flag):
        if flag==0:
            set_values = set(self.nums)
            size = len(self.nums)
        elif flag==1:
            set_values = set(self.stars)
            size = len(self.stars)

        size_set = len(set_values)
        if size != size_set:
            for x in range(size_set, size):
                new_value = input("repeated number, change: ")
                set_values.add(new_value)
            if flag==0:
                self.nums = list(set_values)
            elif flag==1:
                self.stars = list(set_values)
        return True

    def read_values(self):
        result = False
        result2 = False 
        in_nums = input("Enter the numbers :")
        self.nums = in_nums.split(",")

        while result==False or result2==False:
            result = self.check_repeated_values(0)
            result2 = self.check_values(50, self.nums)
        print(self.nums)

        result = False
        result2 = False 
        in_stars = input("Enter the stars :")
        self.stars = in_stars.split(",")

        while result==False or result2==False:
            result = self.check_repeated_values(1)
            result2 = self.check_values(9, self.stars)
        print(self.stars)

    def genarate_random(self):
        self.generated_nums = random.sample(range(1, 50), 5)
        self.generated_stars = random.sample(range(1, 10), 2)     

        self.generated_nums = [str(x) for x in self.generated_nums]
        self.generated_stars = [str(x) for x in self.generated_stars]

        print(self.generated_nums)
        print(list(self.generated_stars))

    def compare(self):
        equal_nums = list(set(self.nums).intersection(set(self.generated_nums)))
        equal_stars = list(set(self.stars).intersection(set(self.generated_stars)))
        print(f"numbers equals {len(equal_nums)}, equal stars {len(equal_stars)}")

if __name__ == "__main__":
    my_euro = euro()
    my_euro.read_values()
    my_euro.genarate_random()
    my_euro.compare()